### PR TITLE
Fflonk

### PR DESCRIPTION
**Pull request by xavi-pinsach**
new feature: fflonk protocol added

### CODE REVIEW

**Comment by phated** [comment]
@xavi-pinsach do we also want to add a "tutorial"-style CI run for fflonk (similar to https://github.com/iden3/snarkjs/blob/master/.github/workflows/tutorial.yml)?

Also, can you please remove the random formatting from this PR? It is making the diff hard to review (feel free to submit a separate formatting PR)

**Comment by xavi-pinsach** [comment]
> @xavi-pinsach do we also want to add a "tutorial"-style CI run for fflonk (similar to https://github.com/iden3/snarkjs/blob/master/.github/workflows/tutorial.yml)?
> 
> Also, can you please remove the random formatting from this PR? It is making the diff hard to review (feel free to submit a separate formatting PR)

What do you mean about 'random formatting from this PR'?

File: cli.js [hunk 1]
```diff
@@ -342,17 +372,17 @@ TODO COMMANDS
...
     } else {
-        return fileName+"."+newExt;
+        return fileName + "." + newExt;
     }
 }
 
 // r1cs export circomJSON [circuit.r1cs] [circuit.json]
 async function r1csInfo(params, options) {
-    const r1csName = params[0] ||  "circuit.r1cs";
+    const r1csName = params[0] || "circuit.r1cs";
```
Comment (phated) [inline cli.js:385]: @xavi-pinsach this is an example of unrelated formatting that you have committed. They are littered throughout the PR. I assume you are using an editor that is formatting on save. Can you please reduce the diff to only the fflonk changes so we don't need to sift through your formatter changes?
